gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -118,9 +125,17 @@ public void addToken(final String token,
   @Override
   public long getTokenExpiration(final String token) throws UnknownTokenException {
     long expiration;
-
-    validateToken(token);
-
+    try {
+      validateToken(token);
+    } catch (final UnknownTokenException e) {
+      /* if token permissiveness is enabled we check JWT token expiration when the token state is unknown */
+      if (permissiveFailureEnabled && StringUtils
```

Comment:
We do the exact same thing twice? Why?

Reply:
I mean we could move this one level up and include it in a different function.

Reply:
Actually moving this code in a separate function would be more untidy (common code in this case would be just `validateToken(token)` and try catch). I don't like the fact that `validateToken(token)` throws an exception instead of returning `true` or `false` that would be a major change to do here. The main function that calculates JWT value `getJWTTokenExpiration(token)` is common for both implementation. Let me know if you still feel strongly about it.

Reply:
The unknown token condition is an important distinction from a known-but-invalid token case; Hence, the exception.